src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -111,10 +115,15 @@ Copyright (c) .NET Foundation. All rights reserved.
                             '$(EnableCompressionInSingleFile)' == 'true' And
                             '$(SelfContained)' != 'true'"
                  ResourceName="CompressionInSingleFileRequiresSelfContained" />
+    <NETSdkError Condition="'$(PublishAot)' == 'true' And
+                            '$(_SkipAotSupportedRidCheck)' != 'true' And
+                            $(_AotSupportedRids.Contains(';$(NETCoreSdkRuntimeIdentifier);')) != 'true'"
```

Comment:
Instead of checking whether a string contains the RuntimeIdentifier, shouldn't we be checking to see whether there is a valid `HostILCompilerPack` or `TargetILCompilerPack` instead for this error?